docs/content/concepts/assets/software-defined-assets.mdx

```diff
@@ -195,6 +195,25 @@ def downstream_asset():
 
 In this example, Dagster doesn’t need to load data from `upstream_asset` to successfully compute the `downstream_asset`. While `downstream_asset` does depend on `upstream_asset`, the key difference with `non_argument_deps` is that data isn’t being passed between the functions. Specifically, the data from the `sugary_cereals` table isn't being passed as an argument to `downstream_asset`.
 
+#### Conditional materialization
+
+In some cases, an asset may not need to be updated in storage each time the decorated function is executed. In this case you can use the `always_materialize` parameter along with `yield` syntax to only materialize an asset to storage in some cases. If an asset is not materialized to storage, downstream assets will not be materialized, and asset sensors monitoring the asset will not trigger
```

Comment:
I think that this can be more direct about what's happening under the hood. Something like "If the `always_materialize` parameter is set to False, you may indicate to the Dagster framework that no data should be persisted to storage by not yielding an output from your computation function. If an output is not emitted during computation, no asset materialization event will be created, the io manager will not be invoked, downstream assets will not be materialized, and asset sensors monitoring the asset will not trigger.